### DISCUSSION TITLE
How to use `KatexMarkdownJS()`

### DISCUSSION THREAD

**Discussion by rleyvasal** [AnswerDotAI/fasthtml#488] [Q&A]
How to use the `KatexMarkdownJS()` function included in `fasthtml`?

I created the following minimal example based on [adv.py](https://github.com/AnswerDotAI/fasthtml/blob/main/examples/adv_app.py), but it is not returning the rendered katex formula

```python
from fasthtml.common import *

# Create fasthtml app
app, rt = fast_app(hdrs=(KatexMarkdownJS('.katex')) )

@rt("/")
def get():
    return Div('This is an inline formula $x =\frac{1}{2}$', cls='katex')

serve()
```

**Comment by dgwyer** [comment]
I think the issue is that `\f` is being treated as a special Python character here. To avoid conflicts like this you can just use a raw string.

```python
from fasthtml.common import *

app, rt = fast_app(live=True,hdrs=[KatexMarkdownJS()])

@rt("/")
def get():
    return (
        Div(r'This is an inline formula $x = \frac {1}{2}$', cls="marked"),
	)

serve()
```

![image](https://github.com/user-attachments/assets/543ba2ea-7c0b-4ea1-b65d-f895e65c26cc)

**Reply by rleyvasal** [reply]
Thanks for the response @dgwyer, but the provided  code still does not render the math formula. I also tried different browsers Edge, Chrome, Chromium, Firefox

![image](https://github.com/user-attachments/assets/52391535-8c4d-4db6-9a8d-2d0bbda8f90c)

**Reply by dgwyer** [reply]
What is the FastHTML code you're using? I managed to get it to render KaTeX using a raw Python string as demonstrated in the screenshot above.

**Reply by rleyvasal** [reply]
Same code you provided, just did copy and paste
![image](https://github.com/user-attachments/assets/1d76bec1-9a5b-4b15-8731-56a8adbef33b)

**Reply by rleyvasal** [reply]
I found the problem, I had an older version of `python-fasthtml`. After upgrade it is working.

Thanks @dgwyer
